pkg/csv2lp/lp_reader.go

```diff
@@ -21,24 +25,35 @@ func LineProtocolFilter(reader io.Reader) *LineProtocolFilterReader {
 	lineReader.LineNumber = 1 // start counting from 1
 	return &LineProtocolFilterReader{
 		lineReader: lineReader,
+		line:       make([]byte, 4096),
+		wrapBuf:    make([]byte, 4096),
 	}
 }
 
 func (state *LineProtocolFilterReader) Read(b []byte) (int, error) {
-	for {
-		bytesRead, err := state.lineReader.Read(b)
-		if err != nil {
-			return bytesRead, err
-		}
-		state.LineNumber = state.lineReader.LastLineNumber
-		buf := b[0:bytesRead]
-		pts, err := models.ParsePoints(buf) // any time precision because we won't actually use this point
-		if err != nil {
-			log.Printf("invalid point on line %d: %v\n", state.LineNumber, err)
-			continue
-		} else if len(pts) == 0 { // no points on this line
-			continue
+	bytesRead, err := state.lineReader.Read(state.line)
+	if err != nil && bytesRead == 0 {
+		return 0, err
+	}
+
+	//read again when we read a partial line at the end of the line reader buffer
+	if bytesRead > 0 && state.line[bytesRead-1] != '\n' {
+		wrapBytesRead, _ := state.lineReader.Read(state.wrapBuf[0:])
```

Comment:
This code ignores any errors from `lineReader.Read`. Errors need to be handled and returned up the stack.